File: leetcode/1513/sol.cpp
class Solution {

    // 1 +2 3 +3 6 +4 10 +5 15 +6 21 +7 28
public:
    int numSub(string s) {
        long long mod = 1e9 + 7;
        auto window = 0;
        auto out = 0;
        auto cur = 0;
        for (auto ch : s) {
            if (ch == '1') {
                window++;
                cur = ((long long) window + cur) % mod;
            } else {
                if (window > 0) {
                    out = ((long long) out + cur) % mod;
                    cur = window = 0;
                }
            }
        }

        return out + cur;
    }
};

File: interview-prep/min_window_substr/sol.py
class Solution:
    def minWindow(self, s: str, t: str) -> str:
        characters = {}

        for c in t:
            if c not in characters:
                characters[c] = 1
            else:
                characters[c] += 1
        
        min_window_size = len(t)

        if len(s) < min_window_size:
            return ""
        
        # init first window
        window = []
        for i in range(min_window_size):
            cur = s[i]
            if not window:
                
            if cur in characters:
                characters[cur] -= 1
            window.append(cur)
            
        # TODO: not implemented yet
        # check if window satisfies the condition


            
            